Doc/c-api/memory.rst

```diff
@@ -150,12 +150,41 @@ The default raw memory block allocator uses the following functions:
 
    Frees the memory block pointed to by *p*, which must have been returned by a
    previous call to :c:func:`PyMem_RawMalloc`, :c:func:`PyMem_RawRealloc` or
-   :c:func:`PyMem_RawCalloc`.  Otherwise, or if ``PyMem_Free(p)`` has been
+   :c:func:`PyMem_RawCalloc`.  Otherwise, or if ``PyMem_RawFree(p)`` has been
    called before, undefined behavior occurs.
 
    If *p* is *NULL*, no operation is performed.
 
 
+.. c:function:: void* PyMem_RawAlignedAlloc(size_t alignment, size_t n)
+
+   Allocates *n* bytes aligned to *alignment* bytes and returns a pointer of
+   type :c:type:`void\*` to the allocated memory, or *NULL* if the request
+   fails.
+
+   *alignment* should be a power of 2 for the best portability.
```

Comment:
Should we enforce C11?

http://en.cppreference.com/w/c/memory/aligned_alloc

"size -  number of bytes to allocate.  An integral multiple of alignment"


`posix_memalign` and `_aligned_malloc` don't care about the multiple  though.

Reply:
It took 17 years to support ISO C99 in CPython 3.6 :-D Python doesn't **have to** enforce anything. I prefer a thin wrapper on top of system allocator.

The question is more on the documentation.

> posix_memalign and _aligned_malloc don't care about the multiple though.

In that case, I suggest to not say anything about the alignment of the *size* argument.